### PR TITLE
Breakpoints

### LINKED ISSUE #155: breakpoints in mipsy_web
We can probably also hook into monaco for this.

### PR DESCRIPTION
**Pull request by SpanishPear**
`break` instruction now works correctly
edit: breakpoints via buttons also work as intended :)
fixes #155

### CODE REVIEW

**Comment by SpanishPear** [comment]
Probably should make it so `Binary` isnt being passed back and forth as we only really need the `binary.breakpoints`

File: crates/mipsy_lib/Cargo.toml [hunk 1]
```diff
@@ -8,12 +8,12 @@ edition = "2021"
...
-serde = { version = "1.0", features = ["derive", "rc"] }  # for data deserialization
-serde_yaml = "0.8"                                        #   - see mips.yaml
-strsim = "0.10"                                           # for error-reporting - string similarity
-linked-hash-map = "0.5.4"                                 # for labels
-colored = "2"                                             # for ansi colors
-strip-ansi-escapes = "0.1"                                # to strip color codes out for strlen calcs
+serde = { version = "1.0", features = ["derive", "rc"] }            # for data deserialization
+serde_yaml = "0.8"                                                  #   - see mips.yaml
+strsim = "0.10"                                                     # for error-reporting - string similarity
+linked-hash-map = { version = "0.5.4", features = ["serde_impl"] }  # for labels
```
Comment (SpanishPear) [inline crates/mipsy_lib/Cargo.toml:14]: for transmission of `Binary` - impl `Serialize, Deserialize`

**Comment by SpanishPear** [comment]
you should be able to play arond with this at 

[the staging server](https://shreys.web.cse.unsw.edu.au/mipsy)

File: crates/mipsy_web/src/worker.rs [hunk 2]
```diff
@@ -78,6 +79,9 @@ pub enum WorkerRequest {
     // The struct that worker can obtain
     CompileCode(FileInformation),
     ResetRuntime(MipsState),
+    UpdateConfig(MipsyWebConfig),
+    // Toggle a breakpoiint at an address
```
Comment (joooooooooooooooooooooooooooooooooooosh) [inline crates/mipsy_web/src/worker.rs:83]:
```suggestion
    // Toggle a breakpoint at an address
```
pwned

Comment (SpanishPear) [reply crates/mipsy_web/src/worker.rs:83]: honestly i saw this and couldnt be bothered so ty 😋

**Review by abiramen — APPROVED** [review]

**Pull request merged**